into/backends/aws.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import print_function, division, absolute_import
+
+import os
+import tempfile
+
+from toolz import memoize
+import boto
+
+from into import discover, CSV, resource, convert
+from ..utils import mkdir_p
+
+
+class _S3(object):
+    pass
+
+
+@memoize
+def S3(cls):
+    """Parametrized S3 bucket Class
+
+    Examples
+    --------
+    >>> import sqlalchemy
+    >>> S3(CSV)
+    S3(CSV)
+    >>> S3(sqlalchemy.Table)
+    """
+    return type('S3(%s)' % cls.__name__, (cls, _S3), {'subtype': cls})
+
+
+@resource.register('s3://.*\.csv', priority=18)
+def resource_s3_csv(uri):
+    return S3(CSV)(uri)
+
+
+@convert.register(CSV, S3(CSV))
+def convert_s3_to_csv(csv, **kwargs):
+    """Download an S3 CSV to a local CSV file."""
```

Comment:
To be more clear.  I actually expect the LHS to turn into `resource('newfile.csv') -> CSV(...)`.  So we don't expect a call to `convert`, we expect a call to `append`.